litentry/core/stf-task/receiver/src/stf_task_receiver.rs

```diff
@@ -15,50 +15,46 @@
 // along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
 
 use crate::{
-	format, AuthorApi, Error, Hash, ShieldingCryptoDecrypt, ShieldingCryptoEncrypt,
-	StfEnclaveSigning, StfTaskContext,
+	format, AuthorApi, Error, HandleState, Hash, SgxExternalitiesTrait, ShardIdentifier,
+	ShieldingCryptoDecrypt, ShieldingCryptoEncrypt, StfEnclaveSigning, StfTaskContext,
 };
+use codec::Decode;
+use ita_sgx_runtime::IdentityManagement;
 use lc_stf_task_sender::{stf_task_sender, RequestType};
 use litentry_primitives::{Assertion, IdentityWebType, Web2Network};
-use log;
+use log::*;
 
 // lifetime elision: StfTaskContext is guaranteed to outlive the fn
-pub fn run_stf_task_receiver<K, A, S>(context: &StfTaskContext<K, A, S>) -> Result<(), Error>
+pub fn run_stf_task_receiver<K, A, S, H>(context: &StfTaskContext<K, A, S, H>) -> Result<(), Error>
 where
 	K: ShieldingCryptoDecrypt + ShieldingCryptoEncrypt + Clone,
 	A: AuthorApi<Hash, Hash>,
 	S: StfEnclaveSigning,
+	H: HandleState,
+	H::StateT: SgxExternalitiesTrait,
 {
 	let receiver = stf_task_sender::init_stf_task_sender_storage()
 		.map_err(|e| Error::OtherError(format!("read storage error:{:?}", e)))?;
 
 	// TODO: When an error occurs, send the extrinsic (error message) to the parachain
+	// TODO: error handling still incomplete, we only print logs but no error handling
+	// TODO: we can further simplify the handling logic
 	loop {
 		let request_type = receiver
 			.recv()
 			.map_err(|e| Error::OtherError(format!("receiver error:{:?}", e)))?;
 
 		match request_type {
-			// TODO: further simplify this
 			RequestType::Web2IdentityVerification(request) =>
 				match lc_identity_verification::web2::verify(request.clone()) {
 					Err(e) => {
-						log::error!("error verify web2: {:?}", e)
+						error!("error verify web2: {:?}", e)
 					},
 					Ok(_) => {
-						match context.create_verify_identity_trusted_call(
-							request.who,
-							request.identity,
-							request.bn,
-						) {
-							Ok(c) =>
-								if let Err(e) = context.submit_trusted_call(&c) {
-									log::error!("submit call(web2 verify_identity) error: {:?}", e)
-								},
-							Err(e) => {
-								log::error!("create call error: {:?}", e)
-							},
-						}
+						let _ = context.decode_and_submit_trusted_call(
+							request.encoded_shard,
+							request.encoded_callback,
+						)?;
```

Comment:
I think we shoud catch and print error here, same as below.

Reply:
> I think we shoud catch and print error here, same as below.

Thanks, I think printing errors is not enough, I have left a TODO comment regarding error handling :(